docs/design/seccomp.md

```diff
@@ -0,0 +1,295 @@
+*  be container-runtime agnostic
+*  allow use of custom profiles
+*  facilitate containerized applications that link directly to libseccomp
+
+## Use Cases
+
+1.  As an administrator, I want to be able to grant access to a seccomp profile
+    to a class of users
+2.  As a user, I want to run an application with a seccomp profile similar to
+    the default one provided by my container runtime
+3.  As a user, I want to run an application which is already libseccomp-aware
+    in a container, and for my application to manage interacting with seccomp
+    unmediated by Kubernetes
+4.  As a user, I want to be able to use a custom seccomp profile and use
+    it with my containers
+
+### Use Case: Administrator access control
+
+Controlling access to seccomp profiles is a cluster administrator
+concern. It should be possible for an administrator to control which users
+have access to which profiles.
+
+The [pod security policy](https://github.com/kubernetes/kubernetes/pull/7893)
+API extension governs the ability of users to make requests that affect pod
+and container security contexts.  The proposed design should deal with
+required changes to control access to new functionality.
+
+### Use Case: Seccomp profiles similar to container runtime defaults
+
+Many users will want to use images that make assumptions about running in the
+context of their chosen container runtime.  Such images are likely to
+frequently assume that they are running in the context of the container
+runtime's default seccomp settings.  Therefore, it should be possible to
+express a seccomp profile similar to a container runtime's defaults.
+
+As an example, all dockerhub 'official' images are compatible with the Docker
+default seccomp profile.  So, any user who wanted to run one of these images
+with seccomp would want the default profile to be accessible.
+
+### Use Case: Applications that link to libseccomp
+
+Some applications already link to libseccomp and control seccomp directly.  It
+should be possible to run these applications unmodified in Kubernetes; this
+implies there should be a way to disable seccomp control in Kubernetes for
+certain containers, or to run with a "no-op" or "unconfined" profile.
+
+Sometimes, applications that link to seccomp can use the default profile for a
+container runtime, and restrict further on top of that.  It is important to
+note here that in this case, applications can only place _further_
+restrictions on themselves.  It is not possible to re-grant the ability of a
+process to make a system call once it has been removed with seccomp.
+
+As an example, elasticsearch manages its own seccomp filters in its code.
+Currently, elasticsearch is capable of running in the context of the default
+Docker profile, but if in the future, elasticsearch needed to be able to call
+`ioperm` or `iopr` (both of which are disallowed in the default profile), it
+should be possible to run elasticsearch by delegating the seccomp controls to
+the pod.
```

Comment:
This example doesn't explain how the application would _further_ limit itself.  the example is about being less restrictive than the default.